code/game/objects/items/twohanded.dm

```diff
@@ -581,10 +589,12 @@
 	throwforce = on ? force_on : initial(force)
 	icon_state = "chainsaw_[on ? "on" : "off"]"
 
-	if(hitsound == "swing_hit")
+	if(on)
 		hitsound = 'sound/weapons/chainsawhit.ogg'
+		AddComponent(/datum/component/butchering, 30, 100, 0, hitsound) //VRRRR
 	else
 		hitsound = "swing_hit"
+		qdel(GetComponent(/datum/component/butchering)) //hard to butcher with a chainsaw that isn't on
```

Comment:
you could set the component's enabled to TRUE/FALSE and checking that in the component's proc instead of creating or deleting it